### DISCUSSION TITLE
Persistent views break after a disconnect

### DISCUSSION THREAD

**Discussion by hypergonial** [Rapptz/discord.py#7437] [Q&A]
Hello!

I've been trying to figure out the cause of this, but it seems that after a longer disconnect from Discord (internet outage) persistent views will simply not work, and require the bot to be restarted. Would it be a good idea to put the view re-adding in `on_ready` instead of a function that only gets called once when the bot is started? I'm not sure on that one.

Thanks!

Edit: I can't mark the answer, seems to be a [Github issue](https://github.com/github/feedback/discussions/5158), if someone else can, it would be greatly appreciated!

**Comment by NCPlayz** [comment]
Seems like it's because discord.py clears all states when it receives READY, which also clears stored views.

**Reply by hypergonial** [reply]
So would it be a good idea to add this code in `on_ready` to fix this?

**Reply by NCPlayz** [reply]
Yeah, that would fix it.
